cluster.go

```diff
@@ -69,7 +70,17 @@ type VM struct {
 	Lifetime  time.Duration
 	PrivateIP string
 	PublicIP  string
-	Zone       string
+	Zone      string
+}
+
+var regionRE = regexp.MustCompile(`(.*[^-])-?[a-z]$`)
+
+func (vm *VM) locality() string {
+	region := "?"
```

Comment:
Perhaps region should be the empty string if there is no region? Not as obvious to the reader, but we almost never view this file.

Reply:
The following isn't a valid locality:

```
--locality=region=,zone=some-zone
```

So: we could either omit it entirely (`--locality=zone=some-zone`), or return an error to the user. Do you have a preference? I have a slight preference for the latter. The more we can assert about our cluster's environment (e.g., "you will _always_ have a region and a zone"), the easier it is to write nightly tests.

Reply:
Ok, makes sense. Carry on.

Reply:
Ok, roachprod now `log.Fatal`s if it can't parse a VM's region. TFTR!